src/sklearndf/classification/wrapper/_wrapper.py

```diff
@@ -105,27 +115,24 @@ def _prediction_with_class_labels(
         delegate_estimator = self.native_estimator
 
         # store the super() object as this is not available within a generator
-        sup = super()
+        _super = cast(ClassifierWrapperDF, super())
 
         # estimators attribute of abstract class MultiOutputEstimator
         # usually the delegate estimator will provide a list of estimators used
         # to predict each output. If present, use these estimators to get
         # individual class labels for each output; otherwise we cannot assign class
         # labels
-        if hasattr(delegate_estimator, "estimators_"):
-            return [
-                sup._prediction_with_class_labels(
-                    X=X, y=output, classes=getattr(estimator, "classes_", None)
-                )
-                for estimator, output in zip(
-                    getattr(delegate_estimator, "estimators_"), y
-                )
-            ]
-        else:
-            return [
-                sup._prediction_with_class_labels(X=X, y=output, classes=None)
-                for output in y
-            ]
+        try:
+            estimators = getattr(delegate_estimator, "estimators_")
+        except AttributeError:
+            return [_super._prediction_with_class_labels(X=X, y=output) for output in y]
```

Comment:
I wanted to ask: Is it a Python convention to manage business logic with exceptions? (as I understand the result is the same). I came across same approach in sklearn (https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/metrics/_scorer.py#L356) and the issue I had with it is that exception messages are a bit less informative: If an exception happens inside `_prediction_with_class_labels` then in the stack trace first we will have `AttributeError` exception that is expected as this is in our business logic and after that a `while handling x another exception occured` with the actual one. 
Does it make a difference?

Reply:
Sure! I pushed commit with such changes.